File: solutions/lua/collatz-conjecture/1/collatz-conjecture.lua
return function(n)
    if n <= 0 then
        error('Only positive numbers are allowed')
    end
    local count = 0
    while n > 1 do
        if n % 2 == 0 then
            n = n / 2
        else
            n = 3 * n + 1
        end
        count = count + 1
    end
    return count
end

File: solutions/typescript/collatz-conjecture/2/collatz-conjecture.ts

export function steps(count: number): number {
    if (!Number.isInteger(count) || count <= 0) {
        throw new Error('Only positive integers are allowed');
    }
    let result = 0;
    while (count > 1) {
        if (count % 2 === 0) {
            count = count / 2;
        } else {
            count = count * 3 + 1;
        }
        result++;
    }
    return result;
}
